components/dropbox/sources/common.mjs

```diff
@@ -35,5 +35,65 @@ export default {
         tz: Date.now(),
       };
     },
+    async isNewFile(update, lastFileModTime) {
+      try {
+        const dpx = await this.dropbox.sdk();
+        let revisions = await dpx.filesListRevisions({
+          path: update.id,
+          mode: {
+            ".tag": "id",
+          },
+          limit: 10,
+        });
+        if (revisions.result) {
+          revisions = revisions.result;
+        }
+        if (revisions.entries.length > 1) {
+          const oldest = revisions.entries.pop();
+          if (lastFileModTime && lastFileModTime >= oldest.client_modified) {
+            return false;
+          }
+        }
+        return true;
+      } catch (err) {
+        console.log(err);
+        throw `Error looking up revisions for file: ${update.name}`;
+      }
+    },
+    async getMediaInfo(update) {
+      try {
+        const dpx = await this.dropbox.sdk();
+        const info = await dpx.filesGetMetadata({
+          path: update.path_lower,
+          include_media_info: true,
+        });
+        if (info.result) {
+          return info.result;
+        }
+        return update;
+      } catch (err) {
+        console.log(err);
+        throw `Error getting media info for file: ${update.name}`;
+      }
+    },
+    async getTemporaryLink(update) {
+      try {
+        const dpx = await this.dropbox.sdk();
+        let response = await dpx.filesGetTemporaryLink({
+          path: update.path_lower,
+        });
+        if (response.result) {
+          response = response.result;
+        }
+        const { link } = response;
+        return link;
+      } catch (err) {
+        if (err?.error?.error_summary?.startsWith("unsupported_file")) {
+          return null;
```

Comment:
Maybe we can add a console log here saying `Unsupported File, NULL returned instead` or something like this?

Reply:
I'm returning null because not all files supports download links, but the source should still emit update events to those files;

just added a comment explaining;